### PR TITLE
Modules remove patch

### PR DESCRIPTION
**Pull request by awgymer**
#1920 

This is an initial commit of an example of how we could achieve this using exisiting functionality to apply a patch in reverse. 

Would need significant improvements in error-handling. 
May be better to do an update on the current `sha`?

## PR checklist

- [ ] This comment contains a description of changes (with reason)
- [ ] `CHANGELOG.md` is updated
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] Documentation in `docs` is updated

### CODE REVIEW

File: nf_core/modules/patch.py [hunk 1]
```diff
@@ -130,3 +130,71 @@ def patch(self, module=None):
...
+        # Try to apply the patch in reverse and move resulting files to module dir
+        temp_module_dir = self.modules_json.try_apply_patch_reverse(
+            module, self.modules_repo.repo_path, patch_relpath, module_relpath
+        )
+        for file in Path(temp_module_dir).glob("*"):
+            file.rename(module_relpath.joinpath(file.name))
+        os.rmdir(temp_module_dir)
+
+        # Remove patch file
+        patch_path.unlink()
```
Comment (mirpedrol) [inline nf_core/modules/patch.py:197]: I would catch possible errors here before removing it from `modules.json`

**Review by mirpedrol — COMMENTED** [review]
Looks very good! :) Maybe we could add a recommendation to lint the module after that? And also an option to force the remove and skip the questionary.

**Comment by awgymer** [comment]
I've added some code to check if the reverted module matches the code for the known `sha` and if it doesn't we suggest a reinstall. 
Still think this process is quite messy and not sure if deleting the patch and removing from the JSON followed by a reinstall would be better?

Not sure what the best way to write a test for this is either.

**Comment by mirpedrol** [comment]
Even if it would be the easiest, I think we should avoid reinstalling because we could lose track from the "installed_by" in modules.json
Regarding tests, we are already testing applying a reverse patch, so I would only test if this function deletes the file and removes it from modules.json.

**Comment by awgymer** [comment]
Failure due to an API rate-limit issue. Tests for 3.9 all passed though. Could I get a re-review on the test I implemented please @mirpedrol

**Pull request merged**